.circleci/config.yml

```diff
@@ -54,9 +54,8 @@ workflows:
       not: << pipeline.parameters.aws_teardown >>
     jobs:
       - godeps
-      - gotest:
-          requires:
-            - godeps
+      - test-race:
+          name: gotest # TODO: Rename to something more descriptive?
```

Comment:
What is this section about? Also can we remove the TODO's?

Reply:
`name: gotest` changes the name of the job as reported in the UI / in webhook calls to GitHub. It's a work-around to allow for naming the job something more descriptive without needing to change our branch-protection rules in GitHub.

I'd like to drop the aliases and update the branch rules to require `test-race` and `lint` instead of `gotest` and `golint`. The rules also need updated to require the arm/windows/mac test jobs. I was planning to hold off on those until I'd landed the last big chunk of CI parallelization-refactor (so there only needs to be one SRE ticket), but I could be convinced to do things incrementally instead.

Reply:
One SRE ticket sounds like a good idea. Can you add some of that info to the TODO?

Reply:
Or to an issue linked from the TODO?